MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.dev.cc

```diff
@@ -0,0 +1,141 @@
+
+#include <iostream>
+#include <fstream>
+#include <Eigen/Core>
+
+using namespace edm;
+using namespace std;
+using namespace ALPAKA_ACCELERATOR_NAMESPACE;
+using namespace ALPAKA_ACCELERATOR_NAMESPACE::magneticFieldParabolicPortable;
+using Vector3f = Eigen::Matrix<float, 3, 1>;
+using Vec3 = CLHEP::Hep3Vector;
+
+struct MagneticFieldKernel {
+  template <typename TAcc, typename T>
+  ALPAKA_FN_ACC void operator()(TAcc const& acc,
+                                Vector3f const* __restrict__ in,
+                                T* __restrict__ out,
+                                size_t size) const {
+    for (auto index : cms::alpakatools::uniform_elements(acc, size)) {
+      out[index*3 + 0] = 0;
+      out[index*3 + 1] = 0;
+      out[index*3 + 2] = magneticFieldAtPoint(in[index]);
+    }
+  }
+};
+
+int main() {
+  // get the list of devices on the current platform
+  auto const& devices = cms::alpakatools::devices<Platform>();
+  if (devices.empty()) {
+    std::cerr << "No devices available for the " EDM_STRINGIZE(ALPAKA_ACCELERATOR_NAMESPACE) " backend, "
+      "the test will be skipped.\n";
+    exit(EXIT_FAILURE);
+  }
+
+  ifstream file;
+  edm::FileInPath mydata("MagneticField/Engine/data/Regression/referenceField_160812_RII_3_8T.bin");
+  file.open(mydata.fullPath().c_str(), ios::binary);
+
+  int count = 0;
+  float px, py, pz;
+  float bx, by, bz;
+  vector<Vector3f> points;
+  vector<GlobalVector> referenceB_vec;
+
+  int numberOfPoints = 100;
+  do {
+    if (!(file.read((char*)&px, sizeof(float)) && file.read((char*)&py, sizeof(float)) &&
+          file.read((char*)&pz, sizeof(float)) && file.read((char*)&bx, sizeof(float)) &&
+          file.read((char*)&by, sizeof(float)) && file.read((char*)&bz, sizeof(float))))
+      break;
+
+    const auto point = Vec3(px, py, pz);
+    if (!isValid(point))
+      continue;
+
+    points.push_back(Vector3f(px, py, pz));
+    referenceB_vec.push_back(GlobalVector(bx, by, bz));
+    count++;
+  } while (count < numberOfPoints);
+
+  const size_t size = points.size();
+  const size_t size_3 = size * 3;
+  // allocate the input and output host buffer in pinned memory accessible by the Platform devices
+  auto in_host = cms::alpakatools::make_host_buffer<Vector3f[], Platform>(size);
+  auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size_3);
+  // fill the input buffers, and the output buffer with zeros
+  for (size_t i = 0; i < size; ++i) {
+    in_host[i] = points[i];
+    for (uint j = 0; j < 3; j++) {
+      out_host[i*3 + j] = 0.;
+    }
+  }
+
+  float resolution = 0.2;
+  float maxdelta = 0.;
+  int fail = 0;
+
+  // run the test on each device
+  for (auto const& device : devices) {
+    auto queue = Queue(device);
+    // allocate input and output buffers on the device
+    auto points_dev = cms::alpakatools::make_device_buffer<Vector3f[]>(queue, size);
+    auto deviceB = cms::alpakatools::make_device_buffer<float[]>(queue, size_3);
```

Comment:
Please use a consistent naming for the variables: `dev` or `device`, before or after the other words, `camelCase` or `snake_case`.

Reply:
(also for the host variables)

Reply:
OK: https://github.com/cms-sw/cmssw/pull/44360/commits/c6936d662ee52343d6446d101f7660a5c8ddffda